src/Microsoft.Health.Dicom.SqlServer/Registration/Experimental/SqlServiceBuilderExtensions.cs

```diff
@@ -0,0 +1,156 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+using Microsoft.Health.Dicom.Core.Features.Store;
+using Microsoft.Health.Dicom.SqlServer.Features.ChangeFeed;
+using Microsoft.Health.Dicom.SqlServer.Features.Common;
+using Microsoft.Health.Dicom.SqlServer.Features.ExtendedQueryTag;
+using Microsoft.Health.Dicom.SqlServer.Features.Query;
+using Microsoft.Health.Dicom.SqlServer.Features.Retrieve;
+using Microsoft.Health.Dicom.SqlServer.Features.Schema;
+using Microsoft.Health.Dicom.SqlServer.Features.Store;
+using Microsoft.Health.Extensions.DependencyInjection;
+using Microsoft.Health.SqlServer.Api.Registration;
+using Microsoft.Health.SqlServer.Features.Schema;
+using Microsoft.Health.SqlServer.Registration;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class SqlServiceBuilderExtensions
+    {
+        public static ISqlServiceBuilder AddIndexDataStores(this ISqlServiceBuilder builder)
+        {
+            IServiceCollection services = EnsureArg.IsNotNull(builder?.Services, nameof(builder));
+
+            services.Add<SqlIndexDataStoreV1>()
+                .Scoped()
+                .AsImplementedInterfaces();
```

Comment:
Add `.AsSelf()` before `.AsImplementedInterfaces()`, there is an Assert checking this https://github.com/microsoft/healthcare-shared-components/blob/af69cf5660398ffe2980dd31cca6443e1ee7d433/src/Microsoft.Health.Extensions.DependencyInjection/TypeRegistrationBuilder.cs#L125.

Reply:
I see. We're actually missing this for our dicom server registration path, so I'll add it there too